src/components/SwapWidget/ConfirmLimitOrderDrawer/index.tsx

```diff
@@ -73,67 +73,73 @@ const ConfirmLimitOrderDrawer: React.FC<Props> = (props) => {
+      return CAVAX[chainId];
+    } else if (inputTokenInfo) {
+      return new Token(
+        inputTokenInfo?.chainId,
+        inputTokenInfo?.address,
+        inputTokenInfo?.decimals,
+        inputTokenInfo?.symbol,
+        inputTokenInfo?.name,
+      );
+    } else if (inputCurrency1) {
+      return new Token(
+        inputCurrency1?.chainId,
+        inputCurrency1?.address,
+        inputCurrency1?.decimals,
+        inputCurrency1?.symbol,
+        inputCurrency1?.name,
+      );
+    } else {
+      return undefined;
+    }
+  };
 
-  const outputCurrency =
-    outputCurrency1 && outputCurrency1?.symbol === CAVAX[chainId].symbol
-      ? CAVAX[chainId]
-      : outputTokenInfo && outputTokenInfo?.symbol === CAVAX[chainId].symbol
-      ? CAVAX[chainId]
-      : outputTokenInfo
-      ? new Token(
-          outputTokenInfo?.chainId,
-          outputTokenInfo?.address,
-          outputTokenInfo?.decimals,
-          outputTokenInfo?.symbol,
-          outputTokenInfo?.name,
-        )
-      : outputCurrency1
-      ? new Token(
-          outputCurrency1.chainId,
-          outputCurrency1.address,
-          outputCurrency1.decimals,
-          outputCurrency1?.symbol,
-          outputCurrency1?.name,
-        )
-      : undefined;
+  const outputCurrency = () => {
+    if (outputCurrency1 && outputCurrency1?.symbol === CAVAX[chainId].symbol) {
+      return CAVAX[chainId];
+    } else if (outputTokenInfo && outputTokenInfo?.symbol === CAVAX[chainId].symbol) {
+      return CAVAX[chainId];
+    } else if (outputTokenInfo) {
+      return new Token(
+        outputTokenInfo?.chainId,
+        outputTokenInfo?.address,
+        outputTokenInfo?.decimals,
+        outputTokenInfo?.symbol,
+        outputTokenInfo?.name,
+      );
+    } else if (outputCurrency1) {
+      return new Token(
+        outputCurrency1?.chainId,
+        outputCurrency1?.address,
+        outputCurrency1?.decimals,
+        outputCurrency1?.symbol,
+        outputCurrency1?.name,
+      );
+    } else {
+      return undefined;
+    }
+  };
 
-  const fiatValueInput = useUSDCPrice(inputCurrency);
-  const fiatValueOutput = useUSDCPrice(outputCurrency);
+  const fiatValueInput = useUSDCPrice(inputCurrency());
+  const fiatValueOutput = useUSDCPrice(outputCurrency());
 
   if (!inputAmount || !outputAmount) return null;
 
   // text to show while loading
   const pendingText = `Submitting order to swap ${trade?.inputAmount?.toSignificant(6)} ${
-    inputCurrency?.symbol
-  } for ${trade?.outputAmount?.toSignificant(6)} ${outputCurrency?.symbol}`;
+    outputCurrency()?.symbol
+  } for ${trade?.outputAmount?.toSignificant(6)} ${outputCurrency()?.symbol}`;
 
   const ConfirmContent = (
     <Root>
       <Header>
         <TokenRow>
-          <CurrencyLogo currency={inputCurrency} size={24} imageSize={48} />
+          <CurrencyLogo currency={inputCurrency()} size={24} imageSize={48} />
```

Comment:
instead of calling `inputCurrency()` and `outputCurrency()`, please call it at the top